genesis/vis/camera.py

```diff
@@ -763,9 +771,18 @@ def is_built(self):
 
     @property
     def idx(self):
-        """The integer index of the camera."""
+        """The global integer index of the camera."""
         return self._idx
 
+    @property
+    @lru_cache(maxsize=1)
```

Comment:
I would rather keep this inside `_batch_render` rather than make it an official property at camera level. This kind of circular dependency is not great.

Besides, I don't think caching the result is relevant for something as simple as finding an index in a list.